typescript/web/src/components/labeling-tool/openlayers-map/openlayers-map.tsx

```diff
@@ -264,7 +267,16 @@ export const OpenlayersMap = () => {
                       console.warn(
                         "Reloading window to prevent bug https://github.com/labelflow/labelflow/issues/431"
                       );
-                      window?.location?.reload?.();
+                      if (
+                        window.localStorage &&
+                        !localStorage.getItem("hasRetriedImageLoad")
+                      ) {
+                        localStorage.hasRetriedImageLoad = true;
+                        window.location.reload();
+                      } else {
+                        localStorage.removeItem("hasRetriedImageLoad");
+                        setImageLoadError(true);
+                      }
```

Comment:
I'm a bit confused about why you need to use both localStorage and queryParams. Couldn't you make it work with only `localStorage.hasRetriedImageLoad` for instance and retrieve its value in `typescript/web/src/components/labeling-tool/labeling-tool.tsx`?

Reply:
The problem with handling that with local storage is that it adds a layer of complexity because we would need to call `localStorage.removeItem` each time we change the image id. Since we are reloading the page once before showing the "Image Not Found" page, we can't really know when the component mounts if the ID of the image is the same as before (so if we are reloading the page to try to load a large image) or if we are arriving on a fresh new image. We don't have that problem by using the queryParams